crates/biome_js_syntax/src/file_source.rs

```diff
@@ -281,10 +281,16 @@ impl JsFileSource {
                 }
             }
             Language::TypeScript { .. } => {
-                if matches!(self.variant, LanguageVariant::Jsx) {
-                    "tsx"
-                } else {
-                    "ts"
+                match self.variant {
+                    LanguageVariant::Standard => "ts",
+                    LanguageVariant::StandardRestricted => {
+                        // This could also be `mts`.
+                        // We choose `cts` because we expect this extension to have more widely used.
+                        // Moreover, it allows more valid syntax such as `import type` with import
+                        // attributes (See `noTypeOnlyImportAttributes` syntax rule).
```

Comment:
```suggestion
                        // This could also be `mts`.
                        // We choose `cts` because we expect this extension to be more widely used.
                        // Moreover, it allows more valid syntax, such as `import type` with import
                        // attributes (See `noTypeOnlyImportAttributes` syntax rule).
```